### ISSUE TITLE
Wine+ddraw part is missing and crashes

### ISSUE THREAD

**Issue by kero990** [ptitSeb/box86#577]
I have already explained the related issues in the issue of ra2, and they are officially submitted to the box86 issue here.
https://github.com/ptitSeb/box86-compatibility-list/issues/44

Chinese users, the hardware configuration is ft2000 cpu (I seem to see its compile option in the latest cmake), when running ra2 and Diablo II with the cnc-ddraw patch, ra2 loses all menu gui elements, including custom campaigns , the faction selection page of the campaign, and opening the settings menu in the game, all will be lost and displayed, but you can operate blindly. The game content is normal, but everything is normal with the cnc loader, but this loader can't play the campaign.

Diablo II is to click on the page where the 4 attributes of the character are added, and the game will crash directly. There is no problem with other sub-pages, and the game itself has no problem.

The above operations have been tested directly with wine on the x86 platform, and also tested on arm64 with exagear+wine, and there are no similar problems, so I guess there are some incompatibilities between box86 and ddraw.

I don't know where the log of box86 is, I can provide the corresponding log under your guidance.
![avatar](https://github.com/kero990/my_html/raw/master/%E6%88%AA%E5%9B%BE_explorer.exe_20220208145858.jpg.jpg)

**Comment by kero990** [comment]
The problem is still
Moreover, after executing BOX86_DYNAREC=0, the running speed of the game drops seriously, and the menu is still invisible. Judging from the sound feedback, it takes 10+ seconds for the interactive behavior of menu clicks to complete, and the speed after entering the game cannot be played normally.
But importantly, the menu is still invisible.

My box86 was integrated into lutris by others, not compiled by myself, so I don't know whether its running environment is normal or not. Is this a bug of box86 or a case of my program.

**Comment by ptitSeb** [comment]
Yes, using `BOX86_DYNAREC=0` makes box86 much slower, o it's not a proprer solution, but it's to understand if it's a problem specific to the Dynarec of box86 or a more generic one.
It seems it's a more genereic one: it will be more difficult to track down.
I need to reproduce the issue on my side first.

**Comment by ptitSeb** [comment]
Is the problem still here or can this ticket be closed?

**Comment by Kippykip** [comment]
Yeah I'm getting weird issues too on a raspberry pi 400.
First I have to run it with the Wine Virtual Desktop otherwise loading screen gets very misaligned.
I set it to 800x600 with winecfg.

What's odd is I'm getting the exact same issue on ddraw wrappers such as DDrawCompat and CNC-ddraw DLL's.

But the games are working fine on Winlator, which incorporates box86 + box64 + wine on Android devices.

This is weird...

**Comment by Kippykip** [comment]
Ah! Figured out something interesting.
Turns out the ddraw wrappers were not even loading at all. It was always prioritising the Wine built-in ddraw DLL's

What you need to do to fix it, is to open a terminal and open **winecfg**.
Go to the **Libraries** tab.
Add a new override for library **ddraw.dll**

Change the selection to Native (Windows). Hit OK

Then copy the cnc-ddraw's ddraw.dll and chuck it into your Red Alert 2's game directory amongst the game.exe and gamemd.exe files.

I noticed a couple main menu items are offset weirdly like the resolution selector.
I also tried this again with the alternative DDrawCompat ddraw.dll included with the steam release, but that just makes the game crash on startup.
